extensions/cornerstone/src/services/ViewportService/CornerstoneViewportService.ts

```diff
@@ -815,6 +849,28 @@ class CornerstoneViewportService extends PubSubService implements IViewportServi
       });
     }
 
+    // When you load SRs that only have measurements that were made in reconstructed views
+    // it's possible that you don't get any volumes. In this case we must grab the volumes
+    // from one of the referenced DS, so that we can activate MPR view without any errors
+    if (volumeInputArray.length === 0) {
```

Comment:
I think you should do it in the loop above 

 const displaySet = displaySetService.getDisplaySetByUID(displaySetInstanceUID);
      if (!volume && displaySet.images) {
        volume = csToolsUtils.getOrCreateImageVolume(displaySet.images.map(image => image.imageId));
      }

Reply:
If I do it in the loop above I'll not be sure that we had no volumes, because some volume might appear in a later iteration. That's why I left it outside the loop.